store/sqlstore/user_store.go

```diff
@@ -367,61 +386,42 @@ func (s SqlUserStore) GetEtagForAllProfiles() store.StoreChannel {
 func (us SqlUserStore) GetAllProfiles(options *model.UserGetOptions) store.StoreChannel {
 	isPostgreSQL := us.DriverName() == model.DATABASE_DRIVER_POSTGRES
 	return store.Do(func(result *store.StoreResult) {
-		var users []*model.User
-		offset := options.Page * options.PerPage
-		limit := options.PerPage
+		query := usersQuery.
+			OrderBy("u.Username ASC").
+			Offset(options.Page * options.PerPage).Limit(options.PerPage)
 
-		searchQuery := `
-			SELECT * FROM Users
-			WHERE_CONDITION
-			ORDER BY Username ASC LIMIT :Limit OFFSET :Offset
-		`
+		query = applyRoleFilter(query, options.Role, isPostgreSQL)
```

Comment:
I gutted `substituteWhereClause` in favour of a simpler `applyRoleFilter` done manually alongside the `options.Inactive` check. `GetAllProfiles` and `GetProfiles` should probably be collapsed with an option anyway.